test/file-translations.spec.ts

```diff
@@ -359,6 +359,81 @@ describe("SmartlingFileTranslationsApi class tests.", () => {
             );
         });
 
+        it("Download translated file with metadata when no headers", async () => {
+            const localeId = "de-DE";
+
+            const downloadMock = {
+                status: 200,
+                arrayBuffer: async (): Promise<ArrayBuffer> => new ArrayBuffer(1),
+                headers: {
+                    get: (): string | null => null
+                }
+            };
+            fileTranslationsApiFetchStub.returns(downloadMock);
+
+            const fileWithMetadata = await fileTranslationsApi.downloadTranslatedFileWithMetadata(
+                accountUid, fileUid, mtUid, localeId);
+
+            sinon.assert.calledOnceWithExactly(
+                fileTranslationsApiFetchStub,
+                `https://test.com/file-translations-api/v2/accounts/${accountUid}/files/${fileUid}/mt/${mtUid}/locales/${localeId}/file`,
+                {
+                    method: "get",
+                    headers: {
+                        Authorization: "test_token_type test_access_token",
+                        "Content-Type": "application/json",
+                        "User-Agent": userAgent
+                    }
+                }
+            );
+
+            assert.ok(fileWithMetadata.contentType === undefined);
+            assert.ok(fileWithMetadata.fileName === undefined);
+            assert.ok(fileWithMetadata.fileContent.byteLength === 1);
+        });
+
+        it("Download translated file with metadata", async () => {
+            const localeId = "de-DE";
+
+            const downloadMock = {
+                status: 200,
+                arrayBuffer: async (): Promise<ArrayBuffer> => new ArrayBuffer(1),
+                headers: {
+                    get: (name: string): string | null => {
+                        switch (name) {
+                        case "content-type":
+                            return "application/xml";
+                        case "content-disposition":
+                            return "attachment; filename=\"test.xml\"";
+                        default:
+                            return null;
+                        }
+                    }
```

Comment:
you could've created a stub (with mocked behavior depending on input) and assigned it here to `get` but this also works I think.

Reply:
re-implemented